### ISSUE TITLE
Kitematic deletes and doesn't recreate a container if the drive access popup timeouts

### ISSUE THREAD

**Issue by Chematronix** [docker/kitematic#3294]
### Expected behavior
When configuring a volume, Kitematic should ask for drive access permission before deleting a container, and not show a deleted container as still existing.

### Actual behavior
In Windows, when setting up a volume, Kitematic deletes the container after one clicks on Enable all Volumes, and *then* it asks for "drive C access"  (with a second popup, that looks like a system notification) if it has never been granted before.  If the popup timeouts (after 10 secs or so), Kitematic keeps showing the deleted container as running, but you can't modify any of its settings.

### Information about the Issue
I don't recall the exact text of the second popup as it doesn't appears since I granted access (and I don't know how to revoke it).  For the same reason, I can't test if the same happens if I cancel it.

### Steps to reproduce the behavior

  1. Create a container with one or more volumes in Kitematic —say, a MariaDB container (set `MYSQL_ROOT_PASSWORD=secret` or so).

  2. In the containers Home tab, click on the /var/lib/mysql volume.
    A popup dialog appears:

>      Enable all volumes to edit files? This may not work with all database containers.

   3. Click Enable all Volumes.  The mariadb container is deleted immediately.

  4.  In Windows, a second popup (that looks like a system notification) appears asking for permissions to the C drive.   Don't click on it, let it timeout and disappear.

  5.  The container is still shown as running in Kitematic, though you can't change any settings, but running `docker ps -a` or restarting Kitematic confirms it's not been recreated.

**Comment by FrenchBen** [comment]
When Kitematic configures a volume, it's actually destroying the old container and re-creating it. 
Since we can't detect the permissions of the shared folder until the container fails, we cannot prevent the user from doing this. Additionally there isn't a wait on the resource from the daemon. 

It is best for the user to check that the folder they're trying to use has the proper permission first, as documented here: https://docs.docker.com/docker-for-windows/#shared-drives

**Comment by Chematronix** [comment]
I won't ask why can't you check which drives are shared, but surely would be better if the user were at least warned of this before hand?  I suspect nowhere in the docs it's mentioned that you will lose your container and its configuration for something like this;  most people don't bother reading the docs thoroughly; and even lovers of RTFM with photographic memory might simply get distracted and miss the popup.

Something along the lines of "Your mission, Admin, should you decide to Accept it, is to share your drive.  As always, should you be caught distracted, Kitematic will disavow any knowledge of your container or its configuration.  Your container will self-destruct in ten seconds.  Good luck, Admin." when Sharing Drives on Demand would do the trick. ;)

A little less aggravating, but leaving a non-functional ghost container in the GUI after failing to recreate it is also confusing.

**Comment by FrenchBen** [comment]
Completely understand where you're coming from - When kitematic was first created, one of the biggest problem was sharing volumes with Virtualbox. By default only the "user" directory was shared in order to provide light guard-rails with what could be shared. 

The concept of updating what seems to be your container, by silently destroying it and re-creating one with the same name has caught a few people off guard. With advances in the Docker product being done by multiple people, kitematic simply lacks the resources it needs to capture all use-cases and better the product. 😞 

TL;DR - If you can't easily detect it when you do a `docker run` then it will be difficult for Kitematic to do this as well.
